pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go

```diff
@@ -305,6 +305,13 @@ func doNetworkFilterOperation(patchContext networking.EnvoyFilter_PatchContext,
 			var retVal *any.Any
 			if userFilter.GetTypedConfig() != nil {
 				// user has any typed struct
+				// The type may not match up exactly. For example, if we use v2 internally but they use v3.
+				// Assuming they are not using deprecated/new fields, we can safely swap out the TypeUrl
```

Comment:
Should we document this assumption?

Reply:
And is it possible to add one test case with a deprecated/new field?

Reply:
I'll see if I can find one deprecated

Reply:
Good catch, I found one and it causes a panic today (probably would fail validation normally, test bypasses this). Added a change to ignore unknown fields

Reply:
Also realized one of our tests I added was actually a v2 proto, switched it back to v3 to get proper coverage